docs/source/configuration/overview.mdx

```diff
@@ -579,6 +579,31 @@ You can configure certain caching behaviors for generated query plans and APQ (b
 - You can configure a Redis-backed _distributed_ cache that enables multiple router instances to share cached values. For details, see [Distributed caching in the GraphOS Router](./distributed-caching/).
 - You can configure a Redis-backed _entity_ cache that enables a client query to retrieve cached entity data split between subgraph reponses. For details, see [Subgraph entity caching in the GraphOS Router](./entity-caching/).
 
+<MinVersion version ="1.49.0">
+
+### Native query planner
+
+</MinVersion>
+
+Starting with v1.49.0, the router can run a Rust-native query planner. This native query planner can be run by itself to plan all queries, replacing the legacy Javascript implementation.
+
+<Tip>
+
+Starting with v2.0.0, to run the most performant and resource-efficient native query planner, set the following options in your `router.yaml`:
+
+```yaml title="router.yaml"
+supergraph:
+  generate_query_fragments: true
+
```

Comment:
let's drop this for this configuration, but mention it as an additional configuration that makes native query planning more optimal?
```suggestion
```